Sources/BraveShared/Extensions/URLExtensions.swift

```diff
@@ -22,4 +23,60 @@ extension URL {
   public var origin: URLOrigin {
     .init(url: self)
   }
+  
+  /// Obtains a clean stripped url from the current Internal URL
+  ///
+  /// Returns the original url without  internal parameters
+  public var stippedInternalURL: URL? {
+    if InternalURL.isValid(url: self),
+       let internalURL = InternalURL(self) {
+      
+      switch internalURL.urlType {
+      case .errorPage:
+        return internalURL.originalURLFromErrorPage
+      case .web3Page, .sessionRestorePage, .readerModePage, .aboutHomePage:
+        return internalURL.extractedUrlParam?.displayURL
```

Comment:
Why does it return `displayURL`?

Reply:
Nice catch you are right `displayURL` is not necessary at all. extractedUrlParam does what we need, I carrie it over from previous implementation, I should have checked.

Fix at